File: 백준/Silver/9461. 파도반 수열/파도반 수열.py
T = [0] * 101
T[1] = 1
T[2] = 1
T[3] = 1
T[4] = 2
T[5] = 2
T[6] = 3
T[7] = 4
T[8] = 5
T[9] = 7
T[10] = 9
N = int(input())
for i in range(1,N+1):
    K = int(input())
    for j in range(11,K+1):
        T[j] = T[j-2]+T[j-3]
    print(T[K])

File: 백준/Silver/9461. 파도반 수열/파도반 수열.js
let readLine = require("readline");

let rl = readLine.createInterface({
  input: process.stdin,
  output: process.stdout,
});
let inputLine = 0;
let T;
rl.on("line", (line) => {
  if (inputLine == 0) {
    T = parseInt(line);
  } else {
    dp = [1, 1, 1, 2, 2, 3, 4, 5, 7, 9];
    let input = parseInt(line);
    if (input < 10) {
      console.log(dp[input - 1]);
    } else {
      for (let i = 10; i <= input; i++) {
        dp[i] = dp[i - 2] + dp[i - 3];
      }
      console.log(dp[input - 1]);
    }
  }
  if (inputLine == T) process.exit();
  inputLine++;
});
